app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -145,118 +184,117 @@ export const snapKeyringBuilder = (
         // If snap is preinstalled and does not request confirmation, skip the confirmation dialog
         const skipConfirmation =
           isSnapPreinstalled(snapId) && !displayConfirmation;
-        // If confirmation dialog is skipped, we consider the account creation to be confirmed
-        let confirmationResult = skipConfirmation;
-        let confirmationApprovalId = '';
-        try {
-          if (!skipConfirmation) {
-            const { id } = controllerMessenger.call(
-              'ApprovalController:startFlow',
-            );
-            confirmationApprovalId = id;
-            confirmationResult = Boolean(
-              await controllerMessenger.call(
-                'ApprovalController:addRequest',
-                {
-                  origin: snapId,
-                  type: SNAP_MANAGE_ACCOUNTS_CONFIRMATION_TYPES.confirmAccountCreation,
-                },
-                true,
-              ),
-            );
-          }
+        // If confirmation dialog are skipped, we consider the account creation to be confirmed until the account name dialog
+        let confirmationResult =
+          skipConfirmation ||
+          (await showAccountCreationDialog(snapId, controllerMessenger));
 
-          if (confirmationResult) {
-            try {
-              await handleUserInput(confirmationResult);
-              await persistKeyringHelper();
-              setSelectedAccountHelper(address);
-              const internalAccount = controllerMessenger.call(
-                'AccountsController:getAccountByAddress',
+        // We need to temporarily add the account to the keyring to show the account name suggestion dialog
+        await handleUserInput(confirmationResult);
+        await persistKeyringHelper();
+        // Start flow to show the account name suggestion dialog
+        const { id: accountNameApprovalId } = controllerMessenger.call(
+          'ApprovalController:startFlow',
+        );
+        confirmationResult = Boolean(
```

Comment:
1. I was wondering if we could make a function out of this? Similar to the `showAccountCreationDialog` one. This way it would be easier to read like `confirmationResult = showAccountNameSuggestionDialog(...)`

2. Shouldn't we check for `if (confirmationResult)` before? Otherwise we would override it, and it might already be cancelled at this point.